### DISCUSSION TITLE
organizzazione fisica - 5

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#84] [Esercizi]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/3b7aef7a-086d-4337-b52b-1f37406b7d89)

**Comment by FrascaHoryzont** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/100992294/abcd6415-48bb-47c9-949b-d6fafa3d1d31)

**Reply by 7Andrei** [reply]
Non manca l'arrotondamento su NBpFI e sul numero accessi? per il resto mi pare tutto giusto

**Reply by bobbylinux** [reply]
in realtà vedo alcune cose che non mi tornano.
il numero di blocchi per il file indice dovrebbe essere parte intera superiore, (i record devono entrarci tutti quindi è necessario arrotondare parte intera superiore e non inferiore, quindi diventa 6956.
Il costo massimo di ricerca è parte intera superiore di log base 2 di 6956 che è 13 + 1 accesso per portare il blocco in memoria e sono 14.

non so poi da dove tiri fuori il c ma devi andare a ritroso se log2(6956) + 1 = 14 devi fare in modo di tornare a trovare il numero di record che è nr <= 4063232, cioè dal numero di record per indice trovi i blocchi del file principale, e torna anche perché se inizialmente hai un costo di 14 ( o 13 secondo i tuoi calcoli) per 3 milioni e 450 mila record, come fai ad avere un numero di record massimo di 2031616 per mantenere quel costo di ricerca? dovrebbe essere maggiore di 3450000.

**Comment by Simone0401** [comment]
Ecco la mia personale soluzione. Ho visto che nella soluzione precedente mancano alcune piccole accortezze che @bobbylinux sottolineava. Spero di non aver fatto errori e che la mia risoluzione (scritta a mano) sia comprensibile.
[Esercizio 5.pdf](https://github.com/user-attachments/files/17015724/Esercizio.5.pdf)
